### ISSUE TITLE
EM image and GT label mismatch

### ISSUE THREAD

**Issue by fgdfgfthgr-fox** [janelia-cellmap/cellmap-segmentation-challenge#127]
![Image](https://github.com/user-attachments/assets/009fe388-4501-4dee-a3f3-346f3e7a7431)

Note the 5, 6, 7 and 8th row. I really don't think the gt label is correct there.

Relevant settings needed to reproduce this issue:
```
request_array_info = {
        "shape": (128, 128, 128),
        "scale": (32, 32, 32),
    }
raw_value_transforms = T.Compose(
        [
            NaNtoNum({"nan": 0, "posinf": None, "neginf": None}),
            Normalize(),
        ],
    )
target_value_transforms = Just default with an additional torch.nan_to_num()
train_loader, val_loader = get_dataloader('datasplit.csv', ["nuc"], 1,
                                              input_array_info=request_array_info, target_array_info=request_array_info, device='cpu',
                                              iterations_per_epoch=8,
                                              train_raw_value_transforms=raw_value_transforms,
                                              val_raw_value_transforms=raw_value_transforms)
classes = ["nuc",]
filter_by_scale = True
```

[datasplit.csv](https://github.com/user-attachments/files/19333572/datasplit.csv)

**Comment by rhoadesScholar** [comment]
Interesting. I see what you mean, but your code is for `nuc` not `mito`.  Also can you give the exact `target_value_transforms`?

That said, I can't help but notice an odd occurrence in the datasplit: `recon-1/labels/groundtruth/crop179/[]`
Normally this should have `nuc` or whatever the requested class(es) are in the brackets, like the other lines in the csv. I doubt this is the cause of the issue, but pushed a fix (hopefully) [here](404e0f37d489ab2e9fe23210610224c9cd0da1ce)

**Comment by fgdfgfthgr-fox** [comment]
> Interesting. I see what you mean, but your code is for `nuc` not `mito`. Also can you give the exact `target_value_transforms`?
> 
> That said, I can't help but notice an odd occurrence in the datasplit: `recon-1/labels/groundtruth/crop179/[]` Normally this should have `nuc` or whatever the requested class(es) are in the brackets, like the other lines in the csv. I doubt this is the cause of the issue, but pushed a fix (hopefully) [here](404e0f37d489ab2e9fe23210610224c9cd0da1ce)

Ah, that's just a naming mistake I made when using the visualization script. Change that from nuc to mito doesn't change the image. And as I said, I just applied `torch.nan_to_num(out['output'].squeeze(0))` to whatever the output from the dataloader. I didn't provide target_value_transforms so it's just the default one.

**Comment by fgdfgfthgr-fox** [comment]
Update to the latest commit and the datasplit no longer has the `recon-1/labels/groundtruth/crop179/[]`
However, the issue still exists.

![Image](https://github.com/user-attachments/assets/b08e19ab-2259-44d9-ba96-9ee48ed1976f)

If you really wonder, this is the code I used:
```
class TrainDataset(torch.utils.data.Dataset):
    def __init__(self, cellmap_d_loader):
        super().__init__()
        self.loader = cellmap_d_loader

    def __len__(self):
        return len(self.loader.loader)

    def __getitem__(self, idx):
        out = self.loader[idx]
        em, gt = out['input'].squeeze(0), torch.nan_to_num(out['output'].squeeze(0))
        return em, gt

if __name__ == "__main__":
    request_array_info = {
        "shape": (128, 128, 128),
        "scale": (32, 32, 32),
    }
    raw_value_transforms = T.Compose(
        [
            NaNtoNum({"nan": 0, "posinf": None, "neginf": None}),
            Normalize(),
        ],
    )
    train_loader, val_loader = get_dataloader('datasplit.csv', ["nuc"], 1,
                                              input_array_info=request_array_info, target_array_info=request_array_info, device='cpu',
                                              iterations_per_epoch=8,
                                              train_raw_value_transforms=raw_value_transforms,
                                              val_raw_value_transforms=raw_value_transforms)
    train_set = Utils_new.TrainDataset(train_loader)
    train_torch_loader = torch.utils.data.DataLoader(dataset=train_set, batch_size=8, num_workers=5, pin_memory=True, persistent_workers=True)
    for i in range(2):
        train_torch_loader_i = iter(train_torch_loader)
        pair = next(train_torch_loader_i)
        inputs = pair[0]
        targets = pair[1]
        outputs = pair[1]
        figs = get_fig_dict(inputs, targets, outputs, ["nuc",])
        figs['nuc'].show()
```

**Comment by rhoadesScholar** [comment]
I see. Can you plot it again without the `nan_to_num`? The ground truth data is padded with NaNs during fetching to allow for accurate loss calculation on crops that are smaller than the requested data view. The [CellMapLossWrapper](https://github.com/janelia-cellmap/cellmap-segmentation-challenge/blob/main/src/cellmap_segmentation_challenge/utils/loss.py) uses these NaNs to appropriately mask the loss. So we shouldn't be seeing the zero padding. 

Still not sure it will look right though...

**Comment by fgdfgfthgr-fox** [comment]
> [@fgdfgfthgr-fox](https://github.com/fgdfgfthgr-fox) Have you used the built in visualization script to look at the data in the Zarrs?

I had, but it's really unintuitive and I cannot spot any issue with the labeling immediately.

**Comment by fgdfgfthgr-fox** [comment]
> I see. Can you plot it again without the `nan_to_num`?

![Image](https://github.com/user-attachments/assets/246678f0-9c79-487f-80c0-d53c3966601e)

Still doesn't look right to me.

**Comment by fgdfgfthgr-fox** [comment]
> [@fgdfgfthgr-fox](https://github.com/fgdfgfthgr-fox) Have you used the built in visualization script to look at the data in the Zarrs?

Also, the sha256sum I got for the matched_res_no_pad.zip I downloaded is `c8a269d582273d5a8619a91555a7c349dc85e0c433be8870e5b8d0044d3e98d9`. In case that help.

**Comment by rhoadesScholar** [comment]
@fgdfgfthgr-fox Can you try again with the updated version of [cellmap-data](https://github.com/janelia-cellmap/cellmap-data/commit/789d818008c1f16ce1e4f4a5799d909c74cf386b)?

**Comment by fgdfgfthgr-fox** [comment]
> [@fgdfgfthgr-fox](https://github.com/fgdfgfthgr-fox) Can you try again with the updated version of [cellmap-data](https://github.com/janelia-cellmap/cellmap-data/commit/789d818008c1f16ce1e4f4a5799d909c74cf386b)?

![Image](https://github.com/user-attachments/assets/21147d6d-0402-4960-b6b9-348eca5aecfc)
Not sure if I should call this a positive change or not.

**Comment by rhoadesScholar** [comment]
This has been a strange one. Just pushed another update to `cellmap_data` that appears to fix it.

**Comment by fgdfgfthgr-fox** [comment]
> This has been a strange one. Just pushed another update to `cellmap_data` that appears to fix it.

![Image](https://github.com/user-attachments/assets/acda5a08-cfcc-488c-93d9-c792acd7fa79)

Seem to be fixed now!
